go/adbc/driver/bigquery/bigquery_database.go

```diff
@@ -115,6 +131,8 @@ func (d *databaseImpl) SetOption(key string, value string) error {
 			d.authType = value
 		case OptionValueAuthTypeUserAuthentication:
 			d.authType = value
+		case OptionValueAuthTypeAppDefaultCredentials:
+			d.authType = value
```

Comment:
Let's consolidate all of these instead of repeating `d.authType = value` over and over